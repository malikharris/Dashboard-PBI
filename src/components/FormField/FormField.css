/* LoginField styles */

.formField {
  height: auto;
  margin-left: auto;
  margin-right: auto;
  margin-top: 30px;
  width: 400px;
  /* border-bottom: 1px solid #bcbdc0; */
}

.formFieldHeader {
  /* Style for "Email" */
  width: 70;
  float: left;
  height: 25px;
  color: #383838;
  font-family: Vitesse, sans-serif;
  font-size: 12px;
  font-weight: 300;
  line-height: 25px;
}

.formFieldPlaceholderInput {
  /* Style for "johndoe@co" */
  width: 400px;
  /* height: 25px; */
  color: #008248;
  float: left;
  font-family: Vitesse;
  font-size: 14px;
  font-weight: 500;
  line-height: 25px;
  /* Style for long bottom border line */
  border: none;
  border-bottom: 1px solid #bcbdc0;
}

::-webkit-input-placeholder {
  /* Edge */
  color: #008248;
}

:-ms-input-placeholder {
  /* Internet Explorer 10-11 */
  color: #008248;
}

::placeholder {
  color: #008248;
}
