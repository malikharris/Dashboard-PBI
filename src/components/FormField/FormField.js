import React from 'react';
import PropTypes from 'prop-types';
import './FormField.css';

const FormField = props => {
  // will need an onChange and onSubmit handler functions, probably
  const { header, value, placeholder, onChange, type } = props;
  return (
    <div className="formField">
      <div className="formFieldHeader">{header || 'Email'}</div>
      <input
        onChange={onChange}
        className="formFieldPlaceholderInput"
        type={type}
        value={value}
        placeholder={placeholder}
      />
    </div>
  );
};

FormField.propTypes = {
  header: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  type: PropTypes.string,
};

FormField.defaultProps = {
  type: 'text',
};

export default FormField;
