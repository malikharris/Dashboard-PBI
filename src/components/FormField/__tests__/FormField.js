import React from 'react';
import { shallow } from 'enzyme';
import FormField from '../FormField';

it('should render correctly', () => {
  const onChange = () => {}; // mocked required function
  const wrapper = shallow(
    <FormField
      onChange={onChange}
      header="Header"
      value="A sample value"
      placeholder="placeholder"
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render with the correct default type of "text"', () => {
  const onChange = () => {}; // mocked required function
  const wrapper = shallow(
    <FormField
      onChange={onChange}
      header="Header"
      value="A sample value"
      placeholder="placeholder"
    />,
  );
  expect(wrapper.find('input').props().type).toEqual('text');
});

it('should render with correct type of "password"', () => {
  const onChange = () => {}; // mocked required function
  const wrapper = shallow(
    <FormField
      onChange={onChange}
      header="Header"
      value="A sample value"
      placeholder="placeholder"
      type="password"
    />,
  );
  expect(wrapper.find('input').props().type).toEqual('password');
});
