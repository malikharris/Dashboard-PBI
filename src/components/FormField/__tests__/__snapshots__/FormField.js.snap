// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="formField"
>
  <div
    className="formFieldHeader"
  >
    Header
  </div>
  <input
    className="formFieldPlaceholderInput"
    onChange={[Function]}
    placeholder="placeholder"
    type="text"
    value="A sample value"
  />
</div>
`;
